File: app/src/main/java/com/example/phonepemoviesapp/viewmodel/MovieViewModel.java
package com.example.phonepemoviesapp.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.example.MovieData;
import com.example.phonepemoviesapp.network.MovieRepository;
import com.example.phonepemoviesapp.util.MainConstant;
import com.google.gson.Gson;

public class MovieViewModel extends AndroidViewModel {
    private Application application;

    private final String TAG = MovieViewModel.class.getSimpleName();
    private final MovieRepository mRepository;
    private MutableLiveData<MovieData> mResponse;

    public MovieViewModel(Application application) {
        super(application);
        mRepository = new MovieRepository(application);
    }

    public LiveData<MovieData> hitApi() {
        String  url = MainConstant.url;
        mResponse = new MutableLiveData<>();
        mRepository.hitApi(url, (stringData, response) -> {
            try {
                MovieData movieResponse = new Gson().fromJson(stringData, MovieData.class);
                mResponse.setValue(movieResponse);
            } catch (Exception e) {
                e.printStackTrace();
                mResponse.setValue(null);
            }
        });
        return mResponse;
    }
}

File: app/src/main/java/com/example/phonepemoviesapp/viewmodel/MovieViewModel.kt
package com.example.phonepemoviesapp.viewmodel

import android.app.Application
import com.example.phonepemoviesapp.network.MovieRepository.hitApi
import androidx.lifecycle.AndroidViewModel
import com.example.phonepemoviesapp.viewmodel.MovieViewModel
import com.example.phonepemoviesapp.network.MovieRepository
import androidx.lifecycle.MutableLiveData
import com.example.phonepemoviesapp.model.MovieData
import androidx.lifecycle.LiveData
import com.example.phonepemoviesapp.util.MainConstant
import com.example.phonepemoviesapp.network.MovieRepository.ISuccessResponseListener
import com.google.gson.JsonObject
import com.google.gson.Gson
import java.lang.Exception

class MovieViewModel(application: Application?) : AndroidViewModel(application!!) {
    private val mRepository: MovieRepository
    private var mResponse: MutableLiveData<MovieData?>? = null
    fun hitApi(): LiveData<MovieData?> {
        val url = MainConstant.url
        mResponse = MutableLiveData()
        mRepository.hitApi(
            url,
            object : ISuccessResponseListener {
                override fun onSuccess(stringData: String?, response: JsonObject?) {
                    try {
                        val movieResponse = Gson().fromJson(stringData, MovieData::class.java)
                        mResponse!!.setValue(movieResponse)
                    } catch (e: Exception) {
                        e.printStackTrace()
                        mResponse!!.setValue(null)
                    }
                }
            })
        return mResponse!!
    }

    init {
        mRepository = MovieRepository(application!!)
    }
}